components/cosmetic_filters/renderer/cosmetic_filters_js_handler.cc

```diff
@@ -465,11 +460,27 @@ void CosmeticFiltersJSHandler::OnHiddenClassIdSelectors(base::Value result) {
         blink::BackForwardCacheAware::kAllow);
   }
 
-  if (!enabled_1st_party_cf_) {
+  if (!enabled_1st_party_cf_)
+    ExecuteObservingBundleEntryPoint();
+}
+
+void CosmeticFiltersJSHandler::ExecuteObservingBundleEntryPoint() {
+  blink::WebLocalFrame* web_frame = render_frame_->GetWebFrame();
+  DCHECK(web_frame);
+
+  if (!bundle_injected_) {
+    static base::NoDestructor<std::string> s_observing_script(
+        LoadDataResource(kCosmeticFiltersGenerated[0].id));
+    bundle_injected_ = true;
     web_frame->ExecuteScriptInIsolatedWorld(
-        isolated_world_id_, blink::WebString::FromUTF8(*g_observing_script),
+        isolated_world_id_, blink::WebString::FromUTF8(*s_observing_script),
         blink::BackForwardCacheAware::kAllow);
   }
+
+  web_frame->ExecuteScriptInIsolatedWorld(
+      isolated_world_id_,
+      blink::WebString::FromUTF8(kObservingScriptletEntryPoint),
+      blink::BackForwardCacheAware::kAllow);
```

Comment:
@atuchin-m do you have any profiler results from a few popular websites how it was before and after? In most cases as I understood we don't didn't do a second injection as there should be a condition to make it. And here I see we call `ExecuteScriptInIsolatedWorld` twice for our most common cases instead of once as before

Reply:
Ok, got your point.
But for many sites like `cnn.com` the script is injected twice.
extra `ExecuteScriptInIsolatedWorld()` with a small script shouldn't make sense , but lets optimize it. Take a look at the last commit.

![cnn](https://user-images.githubusercontent.com/45488748/132870613-1db7bf99-3099-4f0e-8577-f409e7a28a08.png)
example.com before:
![before](https://user-images.githubusercontent.com/45488748/132870628-0dc1d86e-1c9f-4ebd-8038-5e727edf3707.png)
example.com after:
![after](https://user-images.githubusercontent.com/45488748/132870623-4541126f-47c3-4a2c-868c-e09e7e0353fd.png)

Reply:
looks cool, now, thanks!